rs/cli/src/commands/registry.rs

```diff
@@ -125,63 +169,231 @@ impl ExecutableCommand for Registry {
     }
 
     async fn execute(&self, ctx: DreContext) -> anyhow::Result<()> {
-        let writer: Box<dyn std::io::Write> = match &self.output {
-            Some(path) => {
-                let path = path.with_extension("json");
-                let file = fs_err::File::create(path)?;
-                info!("Writing to file: {:?}", file.path().canonicalize()?);
-                Box::new(std::io::BufWriter::new(file))
+        match &self.subcommand {
+            Some(RegistrySubcommand::Diff(diff)) => return self.diff(ctx, diff).await,
+            Some(RegistrySubcommand::Get(get)) => return self.get(ctx, get).await,
+            Some(RegistrySubcommand::History(history)) => return self.history(ctx, history).await,
+            None => {
+                // No subcommand => show help
+                let mut cmd = <Registry as clap::CommandFactory>::command();
+                cmd.print_help()?;
+                Ok(())
             }
-            None => Box::new(std::io::stdout()),
-        };
-
-        // Versions/records mode
-        if self.dump_versions.is_some() {
-            let json_value = self.dump_versions_json(ctx).await?;
-            serde_json::to_writer_pretty(writer, &json_value)?;
-            return Ok(());
         }
+    }
+
+    fn validate(&self, _args: &GlobalArgs, _cmd: &mut clap::Command) {}
+}
+
+impl Registry {
+    async fn get(&self, ctx: DreContext, args: &RegistryGet) -> anyhow::Result<()> {
+        // Resolve version: if negative, use select_versions with range from input to -1, then take first element
+        let height: Option<u64> = if let Some(h) = args.version {
+            if h < 0 {
```

Comment:
Given that height is `Option<u64>` the `Some(h)` cannot be less than 0 because unsigned integers cannot be less than 0 by definition. This should panic and the language should prevent it from happening. The smallest u64 is 0.

I assume this should've been an `i64`?